src/pages/_app.tsx

```diff
@@ -19,11 +20,13 @@ export default function App({ Component, pageProps }: AppProps) {
   return (
     <SessionProvider session={pageProps.session}>
       <CssVarsProvider defaultMode="system">
-        <GarminActivityProvider>
-          <main className={inter.className}>
-            {loading ? <FullPageLoader /> : <Component {...pageProps} />}
-          </main>
-        </GarminActivityProvider>
+        <GarminActivityStorageProvider>
+          <GarminActivityProvider>
```

Comment:
Not sure if this is overkill, we could theoretically combine these contexts, but it felt nice to be able to separate storage from the other methods